### PR TITLE
Add TimescaleDB with AWS Lambda tutorial

### PR DESCRIPTION
**Pull request by zseta**
# Description

Adding the TimescaleDB with AWS Lambda tutorial. (previous discussions happened in [Google docs](https://docs.google.com/document/d/1DkTZkak8-vNYJwVv2oVMGZ1D7KI9lYfI-RnJNwHDkZE/edit#))

https://docs-dev.timescale.com/docs-tutorial-aws-lambda/timescaledb/tutorial-aws-lambda/tutorials/aws-lambda/

### CODE REVIEW

File: timescaledb/tutorials/page-index/page-index.js [hunk 1]
```diff
@@ -162,6 +162,21 @@ module.exports = [
         href: 'simulate-iot-sensor-data',
         excerpt: 'Simulate IoT Sensor Data with TimescaleDB',
       },
+      { title: 'TimescaleDB with AWS Lambda',
+        href: 'aws-lambda',
```
Comment (Loquacity) [inline timescaledb/tutorials/page-index/page-index.js:166]:
```suggestion
        href: 'aws-lambda',
        excerpt: 'Tutorial for using TimescaleDB with AWS Lambda'
```

File: timescaledb/tutorials/aws-lambda/3rd-party-api-ingest.md [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+```json
+[
+  {
+    "Id": "docker_lambda_trigger",
+    "Arn": "<FUNCTION_ARN>"
+  }
+]
+```
+
+Then use the `events put-target` command.
```
Comment (Loquacity) [inline timescaledb/tutorials/aws-lambda/3rd-party-api-ingest.md:201]:
```suggestion
When you have finished, use the `events put-target` command to .... <FIXME>
```

File: timescaledb/tutorials/aws-lambda/3rd-party-api-ingest.md [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+```bash
+aws events put-targets --rule schedule-lambda --targets file://targets.json
+```
+
+To check if the rule is connected to the Lambda function, go to AWS console > Amazon EventBridge > Events > Rules,
```
Comment (Loquacity) [inline timescaledb/tutorials/aws-lambda/3rd-party-api-ingest.md:206]:
```suggestion
To check if the rule is connected to the Lambda function, in the AWS console, navigate to Amazon EventBridge > Events > Rules, and
```

Comment (Loquacity) [reply timescaledb/tutorials/aws-lambda/3rd-party-api-ingest.md:206]: Can probably use a proper right arrow here, too, but that's non-blocking.

File: timescaledb/tutorials/aws-lambda/3rd-party-api-ingest.md [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+]
+```
+
+Then use the `events put-target` command.
+```bash
+aws events put-targets --rule schedule-lambda --targets file://targets.json
+```
+
+To check if the rule is connected to the Lambda function, go to AWS console > Amazon EventBridge > Events > Rules, 
+select your rule and the Lambda function’s name should be under “Target(s)”.
```
Comment (Loquacity) [inline timescaledb/tutorials/aws-lambda/3rd-party-api-ingest.md:207]:
```suggestion
select your rule. The Lambda function’s name is listed under `Target(s)`.
```

**Review by Loquacity — CHANGES_REQUESTED** [review]
Only reviewed the first file, will get back to the second later on.

File: timescaledb/tutorials/aws-lambda/3rd-party-api-ingest.md [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+psycopg2-binary
+pgcopy
+```
+
+Notice how we use `psycopg2-binary` not `psycopg2`. The binary version of the library contains all its dependencies so 
+you don’t need to install them separately.
+
+## Create the Dockerfile
+
+Dockerfile:
```
[the remark this answers is not in the document]
Comment (zseta) [reply timescaledb/tutorials/aws-lambda/3rd-party-api-ingest.md:121]: @Loquacity you mean this whole file, or just the Dockerfile step?

File: timescaledb/tutorials/aws-lambda/create-data-api.md [hunk 6]
```diff
@@ -0,0 +1,302 @@
...
+manually include this library in your deployment package so that it is available to use. We are going to use [Lambda Layers](https://docs.aws.amazon.com/lambda/latest/dg/configuration-layers.html) for this purpose.
+
+What’s a Lambda Layer?
+A Lambda Layer is an archive containing additional code, such as libraries or dependencies. Layers help you use external 
+libraries in your function code that would not be available otherwise.
+
+Let’s add `psycopg2` as a Lambda Layer so the function will be able to use this library.
+
+One issue is that `psycopg2` needs to be built and compiled with statically linked libraries, something that you can't 
+do directly in a Lambda function. A workaround to this issue is to download the [compiled version of the library](https://github.com/jkehler/awslambda-psycopg2) and upload it as a Lambda Layer.
```
[the remark this answers is not in the document]
Comment (zseta) [reply timescaledb/tutorials/aws-lambda/create-data-api.md:20]: https://github.com/timescale/docs/pull/325/commits/c3163bb87acb5aae7dcc2f8237764dd44d5afbd7

File: timescaledb/tutorials/aws-lambda/create-data-api.md [hunk 7]
```diff
@@ -0,0 +1,302 @@
...
+```python
+import os
+config = {'DB_USER': os.environ['DB_USER'],
+          'DB_PASS': os.environ['DB_PASS'],
+          'DB_HOST': os.environ['DB_HOST'],
+          'DB_PORT': os.environ['DB_PORT'],
+          'DB_NAME': os.environ['DB_NAME']}
+```
+
+Now your function code is uploaded along with the database connection details. Let's see if it retrieves the data we expect!
```
[the remark this answers is not in the document]
Comment (zseta) [reply timescaledb/tutorials/aws-lambda/create-data-api.md:159]: the "## Upload the function in AWS Lambda" section uploads the function code.
the "## Add database configuration to AWS Lambda" section uploads the environment variables
both uses the command line tool
the `config = {...}` code is already part of the whole function code that was uploaded previously

File: timescaledb/tutorials/aws-lambda/3rd-party-api-ingest.md (hunk 2 above)
Comment (zseta) [reply timescaledb/tutorials/aws-lambda/3rd-party-api-ingest.md]: https://github.com/timescale/docs/pull/325/commits/6cdfc2d4613581a6d4e9de396f8f812068116c5b

File: timescaledb/tutorials/aws-lambda/3rd-party-api-ingest.md [hunk 8]
```diff
@@ -0,0 +1,221 @@
...
+
+**Required libraries:**
+
+* pandas
+* requests
+* psycopg2
+* pgcopy
+
+If you need to use multiple external libraries - like in this tutorial - your deployment package size can easily be 
+larger than the 250MB limit of Lambda. In this case, you might want to read about. With a Docker container, your
```
[the remark this answers is not in the document]
Comment (zseta) [reply timescaledb/tutorials/aws-lambda/3rd-party-api-ingest.md]: Right, fixed here https://github.com/timescale/docs/pull/325/commits/d87885bad7bc90e6aea287055f83da469694b81c#

**Review by ryanbooz — APPROVED** [review]

File: timescaledb/tutorials/aws-lambda/create-data-api.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (zseta) [reply timescaledb/tutorials/aws-lambda/create-data-api.md:157]: No actually you are 100% right (just realizing now) here's a fix https://github.com/timescale/docs/pull/325/commits/dffa40e43103ee9831586345556dec1f5da11c4c and https://github.com/timescale/docs/pull/325/commits/56ba19d880c7a637f925985f0c6e34b6be60dc08 https://github.com/timescale/docs/pull/325/commits/b5533191032bff9320b88c999af4b5028ada1744 @ryanbooz

**Comment by Loquacity** [comment]
> I'm fine for you to merge if/when you clarify my question. The process/procedure stuff might be a larger conversation that we all need training on. So, I think it's OK as it is now, but definitely want to go back and help this fall into line with the larger standards that Lana is working through.

Yes, I'll make a note to go back over this section and rework it next week.

**Pull request merged**